### ISSUE TITLE
Error with the service "awslogsd"

### ISSUE THREAD

**Issue by ian-tait** [widdix/aws-cf-templates#534]
TemplateID: vpc/vpc-ssh-bastion
Region: us-east-1

When building the bastion, it errors with the following error:

_2021-02-20 08:51:32,484 [DEBUG] Setting service awslogsd to enabled
2021-02-20 08:51:32,488 [ERROR] chkconfig failed with error 1. Output: error reading information on service awslogsd: No such file or directory_

Which in turn sends a failed signal to the AutoScaler causing the CloudForamtion to Rollback.

This seems to be focused around lines 356 on the "vpc-ssh-bastion.yaml"

Has anyone else had this issue?

Below, I have added the full error in the hope it helps:

_2021-02-20 08:51:32,484 [DEBUG] Setting service awslogsd to enabled
2021-02-20 08:51:32,488 [ERROR] chkconfig failed with error 1. Output: error reading information on service awslogsd: No such file or directory

2021-02-20 08:51:32,488 [ERROR] Error encountered during build of awslogs: Could not enable service awslogsd (return code 1)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/construction.py", line 542, in run_config
    CloudFormationCarpenter(config, self._auth_config).build(worklog)
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/construction.py", line 270, in build
    CloudFormationCarpenter._serviceTools[manager]().apply(services, changes)
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/service_tools.py", line 155, in apply
    self._set_service_enabled(service, util.interpret_boolean(serviceProperties["enabled"]))
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/service_tools.py", line 234, in _set_service_enabled
    modifier.set_service_enabled(service, enabled);
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/service_tools.py", line 297, in set_service_enabled
    raise ToolError("Could not %s service %s" % ("enable" if enabled else "disable", service), result.returncode)
ToolError: Could not enable service awslogsd (return code 1)
2021-02-20 08:51:32,491 [ERROR] -----------------------BUILD FAILED!------------------------
2021-02-20 08:51:32,492 [ERROR] Unhandled exception during build: Could not enable service awslogsd (return code 1)
Traceback (most recent call last):
  File "/opt/aws/bin/cfn-init", line 171, in <module>
    worklog.build(metadata, configSets)
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/construction.py", line 129, in build
    Contractor(metadata).build(configSets, self)
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/construction.py", line 530, in build
    self.run_config(config, worklog)
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/construction.py", line 542, in run_config
    CloudFormationCarpenter(config, self._auth_config).build(worklog)
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/construction.py", line 270, in build
    CloudFormationCarpenter._serviceTools[manager]().apply(services, changes)
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/service_tools.py", line 155, in apply
    self._set_service_enabled(service, util.interpret_boolean(serviceProperties["enabled"]))
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/service_tools.py", line 234, in _set_service_enabled
    modifier.set_service_enabled(service, enabled);
  File "/usr/lib/python2.7/dist-packages/cfnbootstrap/service_tools.py", line 297, in set_service_enabled
    raise ToolError("Could not %s service %s" % ("enable" if enabled else "disable", service), result.returncode)
ToolError: Could not enable service awslogsd (return code 1)
2021-02-20 08:51:33,029 [DEBUG] CloudFormation client initialized with endpoint https://cloudformation.us-east-1.amazonaws.com
2021-02-20 08:51:33,029 [DEBUG] Signaling resource AutoScalingGroup in stack ZendeskTicket-BASTION-1M0ND40AKYVY0 with unique ID i-0ba1bbe17f2e919cc and status F
AILU_

**Comment by ian-tait** [comment]
UPDATE:

I have found that the service is called "awslogs" rather than "awslogsd" so I have updated the service name on "vpc-ssh-bastion.yaml"

_2021-02-20 09:57:31,546 [INFO] enabled service awslogs
2021-02-20 09:57:31,546 [DEBUG] Restarting awslogs due to change detected in dependency
2021-02-20 09:57:31,547 [DEBUG] Using service runner: /sbin/service
2021-02-20 09:57:33,585 [INFO] Restarted awslogs successfully_

Now it is erroring on "mariadb"
_2021-02-20 09:57:36,374 [ERROR] mariadb is not available to be installed
2021-02-20 09:57:36,374 [ERROR] Error encountered during build of config: Yum does not have mariadb available for installation_

Why is this happening?

**Comment by michaelwittig** [comment]
Hi @ian-tait 

Do you use the latest version of the templates without modifications? As far as I remember, the service name was renamed from awslogs to awslogsd in Amazon Linux 2 a while ago.

mariadb is 100% not part of our template.

**Comment by ian-tait** [comment]
Hi Michael,

I am using the latest version with some slight changes around tags and naming.
Currently using the latest EC2 "ami-047a51fa27710816e" in us-east-1

Mariadb is part of the bastion build, as see here:
https://github.com/widdix/aws-cf-templates/blob/master/vpc/vpc-ssh-bastion.yaml
Line 432, as it installs packages using cfn-init yum.

**Comment by michaelwittig** [comment]
Oh, know I remember. It installs the mysql client to connect to RDS instances :)

I just created a bastion ssh stack in us-east-1 and it worked. it also worked 3 days ago when the test suite run. can you try to spin up an unmodified template to reproduce your issue? Can you share the parameters of the stack?

**Comment by ian-tait** [comment]
Hi Michael,

After picking through my code, it would seem I found the issue with the AMI; somehow, I was using an older image of an Amazon Linux 1(AL1) machine. Hence the reason for "awslogs" and not "awslogsd". I still haven't figured out how that came about.
I ended up double-checking after my last update, and it was at that point I found I was using the wrong one.

By using the right AMI, this too fixed the issue with the MariaDB. It would appear that this was not available in the repos for AL1 machines.

Not too sure how I got myself in such a mess with my testing but thank you for helping me out.

Thank you
Ian
